### PR TITLE
Update docs for v2 release

### PR DESCRIPTION
**Pull request by sdwheeler**
PowerShell 6x shipped with PSReadline v2 so we had to update the documentation. This PR contains the updated docs that we published. The files have been edited and reformatted to meet our style guidelines and build requirements.

### CODE REVIEW

**Comment by musm** [comment]
Bump. Would be nice to get this in.

File: docs/About/about_PSReadLine.md [hunk 1]
```diff
@@ -0,0 +1,1410 @@
...
+PowerShell console. It provides:
+
+- Syntax coloring of the command line
+- A visual indication of syntax errors
+- A better multi-line experience (both editing and history)
+- Customizable key bindings
+- Cmd and Emacs modes
+- Many configuration options
+- Bash style completion (optional in Cmd mode, default in Emacs mode)
+- Emacs yank/kill ring
```
Comment (SteveL-MSFT) [inline docs/About/about_PSReadLine.md:29]:
```suggestion
- Emacs yank/kill-ring
```

**Review by SteveL-MSFT — APPROVED** [review]
Minor nit

File: docs/Set-PSReadLineOption.md [hunk 2]
```diff
@@ -1,256 +1,274 @@
...
 
-    # RGB value
-    "Command" = "#8181f7"
+ # RGB value
+ "Command" = "#8181f7"
 }
 ```
 
-### --------------  Example 2  --------------
+### Example 2: Set bell style
```
Comment (lzybkr) [inline docs/Set-PSReadLineOption.md:71]: This example is probably not a good example because it might not work on some platforms.

File: docs/Set-PSReadLineOption.md [hunk 3]
```diff
@@ -1,256 +1,274 @@
...
     HistoryNoDuplicates = $true
     HistorySearchCursorMovesToEnd = $true
     Colors = @{
         "Command" = "#8181f7"
     }
 }
-PS C:\> Set-PSReadLineOption @PSReadLineOptions
-```
-
-### --------------  Example 3  --------------
```
Comment (lzybkr) [inline docs/Set-PSReadLineOption.md:80]: Why is this removed?

File: docs/Get-PSReadLineOption.md [hunk 4]
```diff
@@ -1,47 +1,109 @@
...
+multiple kinds of tokens.
 
-The object returned can be used to change PSReadLine options.
+## EXAMPLES
 
-This provides a slightly simpler way of setting syntax coloring options for multiple kinds of tokens.
+### Example 1: Get options and their values
 
-## EXAMPLES
+```powershell
```
Comment (lzybkr) [inline docs/Get-PSReadLineOption.md:34]: If this example is included, there should be a note that the example output is dependent on which version of PowerShell you use. The *Color options look different if you run this in Windows PowerShell.

File: docs/Get-PSReadLineOption.md [hunk 5]
```diff
@@ -1,47 +1,109 @@
...
 
-You cannot pipe objects to Get-PSReadLineOption
+You cannot pipe objects to this cmdlet.
 
 ## OUTPUTS
 
 ### Microsoft.PowerShell.PSConsoleReadLineOptions
 
-An instance of the current options.
-Changing the values will update the settings in PSReadLine directly without invoking Set-PSReadLineOption.
```
Comment (lzybkr) [inline docs/Get-PSReadLineOption.md:41]: Why is this removed? This is an important and useful detail.

File: docs/PSConsoleHostReadline.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+## SYNTAX
+
+```
+PSConsoleHostReadLine
+```
+
+## DESCRIPTION
+
+`PSConsoleHostReadLine` is the main entry point for the PSReadLine module. The PowerShell console
+host automatically loads the PSReadLine modules and calls this function. Under normal operating
```
Comment (lzybkr) [inline docs/PSConsoleHostReadline.md:24]: PSReadLine module

Comment (lzybkr) [reply docs/PSConsoleHostReadline.md:24]: automatically loads in interactive sessions. The module is not loaded for non-interactive sessions, e.g. `pwsh -File foo.ps1`.

File: docs/PSConsoleHostReadline.md [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+conditions, this function is not intended to be used from the command line.
+
+If you wanted to create your own custom ReadLine function, you could replace this function with
```
Comment (lzybkr) [inline docs/PSConsoleHostReadline.md:27]: I wouldn't really say here "if you wanted to create your own ..." but you could refine the description so that is inferred.

This module doesn't enable creating your own custom ReadLine function, it is a custom ReadLine function.

The extension point `PSConsoleHostReadLine` is special to the console host - it will call any alias/function/script with this name. PSReadLine defines this function so that it is called from the console host.

File: docs/Remove-PSReadlineKeyHandler.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+## DESCRIPTION
+
+The `Remove-PSReadLineKeyHandler` cmdlet removes a specified key binding.
+
+## EXAMPLES
+
+### Example 1: Remove a binding
+
+```powershell
+Remove-PSReadLineKeyHandler -Chord Ctrl+Shift+B
```
Comment (lzybkr) [inline docs/Remove-PSReadlineKeyHandler.md:32]: While V1 required `Shift`, V2 does not for normal keys like `B` or even `!`. In V1, you might have used `Ctrl+Shift+1` or `Ctrl+Shift+b`, in V2 you would use `Ctrl+!` or `Ctrl+B`. Also note that `Ctrl+b` is not the same as `Ctrl+B`.

This is worth noting because it is a minor breaking change.

File: docs/Remove-PSReadlineKeyHandler.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+```
+
+This command removes the binding from the key combination, or chord, `Ctrl+Shift+B`.
```
Comment (lzybkr) [inline docs/Remove-PSReadlineKeyHandler.md:35]: See previous comment about `Shift`.

File: docs/Remove-PSReadlineKeyHandler.md [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+The `Ctrl+Shift+B` chord is created in the `Set-PSReadLineKeyHandler` article.
```
Comment (lzybkr) [inline docs/Remove-PSReadlineKeyHandler.md:36]: See previous comment about `Shift`.

File: docs/Remove-PSReadlineKeyHandler.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+## PARAMETERS
+
+### -Chord
+
+Specifies an array of keys or sequences of keys to be removed. A single binding is specified by
+using a single string. If the binding is a sequence of keys, separate the keys by a comma, as in
+the following example:
+
+`Ctrl+X,Ctrl+L`
```
Comment (lzybkr) [inline docs/Remove-PSReadlineKeyHandler.md:46]: See previous comment about `Shift` - this example should probably be `Ctrl+x,Ctrl+l`.

File: docs/Set-PSReadLineKeyHandler.md [hunk 12]
```diff
@@ -1,150 +1,157 @@
...
 
 ## SYNTAX
 
-### Set 1
+### ScriptBlock
 
 ```
-Set-PSReadLineKeyHandler [-Chord] <String[]> [-ScriptBlock] <ScriptBlock> [-BriefDescription <String>]
- [-Description <String>] [-ViMode <ViMode>]
+Set-PSReadLineKeyHandler [-ScriptBlock] <ScriptBlock> [-BriefDescription <String>] [-Description <String>]
```
Comment (lzybkr) [inline docs/Set-PSReadLineKeyHandler.md:22]: Why are the parameters reordered here? This doesn't match the `Get-Command -Syntax` output which I think is better than this change.

File: docs/Set-PSReadLineKeyHandler.md [hunk 13]
```diff
@@ -1,150 +1,157 @@
...
+ [-Chord] <String[]> [-ViMode <ViMode>] [<CommonParameters>]
 ```
 
-### Set 2
+### Function
 
 ```
-Set-PSReadLineKeyHandler [-Chord] <String[]> [-Function] <String> [-ViMode <ViMode>]
+Set-PSReadLineKeyHandler [-Chord] <String[]> [-ViMode <ViMode>] [-Function] <String> [<CommonParameters>]
```
Comment (lzybkr) [inline docs/Set-PSReadLineKeyHandler.md:29]: Again, why reorder?

File: docs/Set-PSReadLineKeyHandler.md [hunk 14]
```diff
@@ -1,150 +1,157 @@
...
 
 ## EXAMPLES
 
-### --------------  Example 1  --------------
+### Example 1: Bind the arrow key to a function
 
-```
-PS C:\> Set-PSReadLineKeyHandler -Key UpArrow -Function HistorySearchBackward
+This command binds the up arrow key to the function **HistorySearchBackward**. This function uses
+the current contents of the command line as the search string used to search the command history.
```
Comment (lzybkr) [inline docs/Set-PSReadLineKeyHandler.md:43]: This function searches command history for command lines that **start with** the current contents of the command line.

File: docs/Set-PSReadLineKeyHandler.md [hunk 15]
```diff
@@ -1,150 +1,157 @@
...
+```powershell
+Set-PSReadLineKeyHandler -Chord UpArrow -Function HistorySearchBackward
 ```
 
-This command binds the up arrow key to the function HistorySearchBackward which will use the currently entered command line as the beginning of the search string when searching through history.
+### Example 2: Bind a key to a script block
 
-### --------------  Example 2  --------------
+This example shows how a single key can be used to run a command. The command binds the key
+`Ctrl+Shift+B` to a script block that clears the line, inserts the word "build", and then accepts
```
Comment (lzybkr) [inline docs/Set-PSReadLineKeyHandler.md:52]: See previous comment about `Shift`.

File: docs/Set-PSReadLineKeyHandler.md [hunk 16]
```diff
@@ -1,150 +1,157 @@
...
 }
 ```
 
-This example binds the key Ctrl+Shift+B to a script block that clears the line, inserts build, then accepts the line.
-This example shows how a single key can be used to execute a command.
-
 ## PARAMETERS
 
-### -Chord
+### -BriefDescription
```
Comment (lzybkr) [inline docs/Set-PSReadLineKeyHandler.md:65]: Why are the parameters reordered here? I won't merge this PR with reordered content like this - it's too hard to review and too easy to miss things, e.g. you've removed at least one parameter.

File: docs/About/about_PSReadLine.md [hunk 17]
```diff
@@ -0,0 +1,1410 @@
+---
+ms.date:  12/08/2018
+schema:  2.0.0
+locale:  en-us
+keywords:  powershell,cmdlet
+title:  About PSReadLine
+---
+# PSReadLine
```
Comment (lzybkr) [inline docs/About/about_PSReadLine.md:8]: There are some big problems here:

* I can't tell what you've changed because this file isn't a renamed version of `about_PSReadLine.help.txt`.
* You've extracted the XML doc comments from the source? The XML doc comments will likely not be kept in sync with this file without automation.
* And speaking of automation - the build script for the module generates the bulk of this file from the XML. That automation also verifies all bindable functions are documented. Your change isn't detected by that automation because of the filename change.

**Review by lzybkr — CHANGES_REQUESTED** [review]
If you need to make build automation changes to accommodate some of these documentation changes, please open an issue and work with @daxian-dbw.

File: docs/Get-PSReadLineOption.md (hunk 4 above)
Comment (sdwheeler) [reply docs/Get-PSReadLineOption.md:34]: The purpose of the PR was to update the documentation for PSReadLine v2. If you install PSReadLine v2 on Windows PowerShell 5.1 does it work differently?

File: docs/PSConsoleHostReadline.md (hunk 6 above)
Comment (sdwheeler) [reply docs/PSConsoleHostReadline.md:24]: I will rework this description and submit an update.

File: docs/PSConsoleHostReadline.md (hunk 7 above)
Comment (sdwheeler) [reply docs/PSConsoleHostReadline.md:27]: I will rework this description and submit an update.

File: docs/Remove-PSReadlineKeyHandler.md (hunk 8 above)
Comment (sdwheeler) [reply docs/Remove-PSReadlineKeyHandler.md:32]: Will update.

File: docs/Remove-PSReadlineKeyHandler.md (hunk 9 above)
Comment (sdwheeler) [reply docs/Remove-PSReadlineKeyHandler.md:35]: Will update

File: docs/Remove-PSReadlineKeyHandler.md (hunk 10 above)
Comment (sdwheeler) [reply docs/Remove-PSReadlineKeyHandler.md:36]: Will update

File: docs/Remove-PSReadlineKeyHandler.md (hunk 11 above)
Comment (sdwheeler) [reply docs/Remove-PSReadlineKeyHandler.md:46]: Will fix

File: docs/Set-PSReadLineKeyHandler.md (hunk 12 above)
Comment (sdwheeler) [reply docs/Set-PSReadLineKeyHandler.md:22]: Not sure what happened here. Will fix this.

File: docs/Get-PSReadLineOption.md (hunk 4 above)
Comment (lzybkr) [reply docs/Get-PSReadLineOption.md:34]: The output will look different, yes, you'll see the following in Windows PowerShell:

```CommandColor                           : "$([char]0x1b)[93m"
CommentColor                           : "$([char]0x1b)[32m"
ContinuationPromptColor                : "$([char]0x1b)[37m"
DefaultTokenColor                      : "$([char]0x1b)[37m"
EmphasisColor                          : "$([char]0x1b)[96m"
ErrorColor                             : "$([char]0x1b)[41;93m"
KeywordColor                           : "$([char]0x1b)[92m"
MemberColor                            : "$([char]0x1b)[97m"
NumberColor                            : "$([char]0x1b)[97m"
OperatorColor                          : "$([char]0x1b)[90m"
ParameterColor                         : "$([char]0x1b)[90m"
SelectionColor                         : "$([char]0x1b)[30;47m"
StringColor                            : "$([char]0x1b)[36m"
TypeColor                              : "$([char]0x1b)[37m"
VariableColor                          : "$([char]0x1b)[92m"
```

Note the `$([char]0x1b)` instead of `` `e ``.

**Comment by sdwheeler** [comment]
Stale PR - closing
